tests/test_main_shutdown.py

```
@@ -0,0 +1,952 @@
+    # Should raise the Matrix connection exception
+    with (
+        patch(
+            "mmrelay.main.asyncio.get_running_loop",
+            side_effect=_make_patched_get_running_loop(),
+        ),
+        patch(
+            "mmrelay.main.asyncio.to_thread",
+            side_effect=inline_to_thread,
+        ),
+    ):
+        with pytest.raises(Exception) as exc_info:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)